chemprop/models/utils.py

```diff
@@ -0,0 +1,30 @@
+from os import PathLike
+import torch
+from sklearn.preprocessing import StandardScaler
+
+from chemprop.models.model import MPNN
+from chemprop.models.multi import MulticomponentMPNN
+
+def save_model(path: PathLike, model: MPNN, input_scalers: list[StandardScaler] | None, output_scaler: StandardScaler | None) -> None:
+    torch.save(
+        {
+            "hyper_parameters": model.hparams,
+            "state_dict": model.state_dict(),
+            "input_scalers": input_scalers,
+            "output_scaler": output_scaler,
+        },
+        path,
+    )
+
+
+def load_model(path: PathLike) -> tuple[MPNN, list[StandardScaler] | None, StandardScaler | None]:
+    try:
+        model = MulticomponentMPNN.load_from_file(path)
+    except KeyError: # MPNN
+        model = MPNN.load_from_file(path)
```

Comment:
I don't like doing something based on try/except logic. Is there some way we can think of embedding the model class into the saved file? Possibly including a `'cls'` key that contains `model.__class__` and just loading that?

Reply:
or you can just supply `multicomponent` as an input argument and that'd probably work better.

Reply:
I like having `multicomponent` as the input argument to make it explicit! Will do that.